### PR TITLE
improvements to uber

### PR DESCRIPTION
**Pull request by mclarsen**

### CODE REVIEW

File: bootstrap-env.sh [hunk 1]
```diff
@@ -63,7 +63,7 @@ function info
 
 function uberenv
 {
-    python scripts/uberenv/uberenv.py "$ALL_ARGS"
+    python scripts/uberenv/uberenv.py "$@"
```
Comment (mclarsen) [inline bootstrap-env.sh]: This is the proper way to forward args. Not sure why the other method did not work

File: scripts/uberenv/packages/uberenv-ascent/package.py [hunk 2]
```diff
@@ -68,8 +68,6 @@ class UberenvAscent(Ascent):
     # in upstream spack package
     depends_on("cmake@3.8.2:3.9.999", when="+cmake")
 
-    # use old version of mpi4py to avoid build issues with cython
-    depends_on("py-mpi4py@2.0.0:2.9.999", when="+mpi")
```
Comment (mclarsen) [inline scripts/uberenv/packages/uberenv-ascent/package.py:72]: moving this into the main ascent pack package

File: scripts/uberenv/spack_configs/darwin/packages.yaml [hunk 3]
```diff
@@ -27,7 +27,7 @@ packages:
       lapack: [openblas]
       mkl: [intel-mkl]
       mpe: [mpe2]
-      mpi: [mpich, openmpi]
+      mpi: [openmpi, mpich]
```
Comment (mclarsen) [inline scripts/uberenv/spack_configs/darwin/packages.yaml:30]: My system does not like the mpich patch

**Comment by mclarsen** [comment]
openblas is timing out in travis

File: scripts/uberenv/spack_configs/darwin/packages.yaml (hunk 3 above)
Comment (cyrush) [reply scripts/uberenv/spack_configs/darwin/packages.yaml:30]: I added a bug report for the patch issue with mpich 3.2.1:
https://github.com/spack/spack/issues/8432

I am having bad luck with openmpi -- I suggest we switch back and prefer another version of mpich - 3.2 (sans .1) which will patch, or even older 3.0.4 should work.

**Comment by mclarsen** [comment]
I am going to merge these changes in with the vtkm update.

**Pull request closed without merging**